lsp/nls/src/requests/completion.rs

```diff
@@ -21,42 +104,149 @@ pub fn handle_completion(
         .id_of(params.text_document_position.text_document.uri.as_str())
         .unwrap();
 
+    let text = server.cache.files().source(file_id);
     let start = position_to_byte_index(
         server.cache.files(),
         file_id,
         &params.text_document_position.position,
     )
     .unwrap();
 
-    let locator = (file_id, ByteIndex(start as u32));
+    let locator = (file_id, ByteIndex((start - 1) as u32));
```

Comment:
Why `- 1`? Maybe this should be documented, because future readers might ask the same question

Reply:
`-1` because at the current cursor position, there is no *linearization* item there, so we try to get the item just before the cursor.